lua/plenary/path.lua

```diff
@@ -202,20 +202,31 @@ function Path:mkdir(opts)
   opts = opts or {}
 
   local mode = opts.mode or 448 -- 0700 -> decimal
-  local parents = F.if_nil(opts.parents, false)
-  local exists_ok = F.if_nil(opts.exists_ok, true)
+  local parents = F.if_nil(opts.parents, false, opts.parents)
+  local exists_ok = F.if_nil(opts.exists_ok, true, opts.exists_ok)
 
   if not exists_ok and self:exists() then
     error("FileExistsError:" .. self:absolute())
   end
 
   if not uv.fs_mkdir(self:absolute() or self.filename,  mode) then
     if parents then
-      -- TODO: Find all the parents
-      error("Not implemented")
+      local dirs = vim.split(self:absolute(), self._sep)
+      local processed = ''
+      for _, dir in ipairs(dirs) do
+        if dir ~= '' then
+          local joined = processed .. self._sep .. dir
+          if uv.fs_mkdir(joined, mode) then
+            processed = joined
+          else
+            -- Directory might already exists but we still wanna keep going
```

Comment:
Maybe i need to add a comment explaining it but we can't start creating files at cwd.
Because maybe i wanna do `Path:new('/home/conni/asdf'):mkdir({parents = true})` and my cwd is something completely different. So i have to just try creating new files and if it fails we need to continue because maybe the current dir already exists but the next one not. Does that make sense? Or do you have another idea

Reply:
I'd say split the path, take the last item and add it to local var and for loop with if body `vim.loop.fs_stat(var) == nil then mkdir

Reply:
Okay but what about this: `/home/conni/asdf/asdf2/asdf3/asdf4` assuming only `/home/conni` exists.
With your method i need to try to create `asdf4` fails then `asdf3` fails then `asdf2` then `asdf1` success and then go the other way around. So next is `asdf2`, etc...

Also running `fs_mkdir` on an already existing file is not worse then checking exists(). But if i check exists() on an non existing item and then create i basically do a unnecessary step. 🤔

Reply:
`lua print(vim.loop.cwd())`